src/dcd/server/autocomplete/ufcs.d

```diff
@@ -169,3 +173,32 @@ bool doUFCSSearch(string beforeToken, string lastToken)
     // we do the search if they are different from eachother
     return beforeToken != lastToken;
 }
+
+void lookupUFCSForDotChaining(T)(Scope* completionScope, ref T tokens, size_t cursorPosition, ref AutocompleteResponse response)
+{
+    auto filteredTokens = filter!(t => !t.text.empty)(tokens);
+
+    if (filteredTokens.empty)
+        return;
+
+    auto foundTokens = filteredTokens.array;
+    auto possibleUFCSSymbols = completionScope.getSymbolsByNameAndCursor(istring(null), cursorPosition);
+    for (auto i = foundTokens.length; i > 0; i--)
+    {
+        // Finding function return value type
+        auto foundSymbols = filter!(s => s.name == foundTokens[i - 1].text && s.kind == CompletionKind
+                .functionName)(possibleUFCSSymbols);
```

Comment:
UFCS resolution and call-chains should be done on the dsymbol side. Otherwise we would need to resolve templates and other things in the future on the DCD side, which is quite bad architecturally.

Reply:
So you are suggesting that this function should be placed somewhere within dsymbol?